docs/src/pages/demos/selects/MultipleSelect.tsx

```diff
@@ -0,0 +1,200 @@
+import React from 'react';
+import clsx from 'clsx';
+import { makeStyles, useTheme, Theme } from '@material-ui/core/styles';
+import Input from '@material-ui/core/Input';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+import FormControl from '@material-ui/core/FormControl';
+import ListItemText from '@material-ui/core/ListItemText';
+import Select from '@material-ui/core/Select';
+import Checkbox from '@material-ui/core/Checkbox';
+import Chip from '@material-ui/core/Chip';
+
+const useStyles = makeStyles((theme: Theme) => ({
+  root: {
+    display: 'flex',
+    flexWrap: 'wrap',
+  },
+  formControl: {
+    margin: theme.spacing(1),
+    minWidth: 120,
+    maxWidth: 300,
+  },
+  chips: {
+    display: 'flex',
+    flexWrap: 'wrap',
+  },
+  chip: {
+    margin: 2,
+  },
+  noLabel: {
+    marginTop: theme.spacing(3),
+  },
+}));
+
+const ITEM_HEIGHT = 48;
+const ITEM_PADDING_TOP = 8;
+const MenuProps = {
+  PaperProps: {
+    style: {
+      maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
+      width: 250,
+    },
+  },
+};
+
+const names = [
+  'Oliver Hansen',
+  'Van Henry',
+  'April Tucker',
+  'Ralph Hubbard',
+  'Omar Alexander',
+  'Carlos Abbott',
+  'Miriam Wagner',
+  'Bradley Wilkerson',
+  'Virginia Andrews',
+  'Kelly Snyder',
+];
+
+type ValuesOf<T extends string[]> = T[number];
+type Name = ValuesOf<typeof names>;
```

Comment:
why not just use `string` and `string[]`?

Reply:
This was just a bit of future thinking for when const assertions are available as names is basically an enum here so using a value not in names doesn't make any sense.

Currently it's unnecessary and excessive typing as it just widens straight to `string` and `string[]`, I just did a bit of testing with using names as a read-only array using typescript 3.4 and the editor integration was awesome.

Reply:
Ah I see! That's really cool!

Reply:
I would rather keep the demos as simple as possible. In this demo we don't gain much by restricting the names to literal unions instead of just strings. There's no runtime error this prevents. Since we cast the value later anyway the safety added here is lost anyway.